src/build/__main__.py

```diff
@@ -33,13 +33,8 @@
     'underline': '\33[4m',
     'reset': '\33[0m',
 }
-
-
-def _print(message: str) -> None:
-    color = sys.stdout.isatty() and 'NO_COLOR' not in os.environ
-    for name, code in _STYLES.items():
-        message = message.replace(f'[{name}]', code if color else '')
-    print(message)
+if 'FORCE_COLOR' not in os.environ and not sys.stdout.isatty() or 'NO_COLOR' in os.environ:
```

Comment:
This is for me is hard to read and reason about :thinking: Can we do instead something like:

```python
def _init_colors():
	  use_colors = sys.stdout.isatty()
	  if os.environ.get('NO_COLOR', '').strip():
	     use_colors = False
	  elif os.environ.get('FORCE_COLOR', '').strip():
	     use_colors = True
	  
	  styles = {..}
	  if use_colors = False:
	     styles = {color: '' for color in _STYLES}
      return styles

_STYLES = _init_colors
```

This will make testing it easier to. No need to do the importlib reload tricks.

Reply:
I went with

```python
def _init_colors() -> Dict[str, str]:
    if 'NO_COLOR' in os.environ:
        if 'FORCE_COLOR' in os.environ:
            warnings.warn(
                'Both NO_COLOR and FORCE_COLOR environment '
                'variables are set, disabling color'
            )
        return _NO_COLORS
    elif 'FORCE_COLOR' in os.environ or sys.stdout.isatty():
        return _COLORS
    return _NO_COLORS


_STYLES = _init_colors()
```

I don't really consider the module reload a trick, in fact, it helps us make sure the styles are set correctly at load :stuck_out_tongue:

Reply:
ok 👍 I still don't like that we consider empty value as set (as @henryiii pointed out that's in our opinion a bad practice, but 🤷 )